src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,246 @@
+		const propertyFilter = data.query.propertyFilter || '-all-'; // '-all-' for all properties OR ccm-stuff
+		const searchWord = data.query.searchQuery || 'img'; // will give pictures of flowers as default
+
+		// const filterOptions = data.query.filterOptions
+
+
+		if (!this.checkEnv()) {
+			return 'Update your env variables. See --> src/services/edusharing/envTemplate';
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			url: `${process.env.ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?contentType=${contentType}&skipCount=${skipCount}&maxItems=${maxItems}&sortProperties=${sortProperties}&sortProperties=cm%3Amodified&sortAscending=${sortAscending}&sortAscending=false&propertyFilter=${propertyFilter}&`,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] }
+				],
+				facettes: ['cclom:general_keyword']
+			}),
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		/* const options2 = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			//url: `${process.env.ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?contentType=${contentType}&skipCount=${skipCount}&maxItems=${maxItems}&sortProperties=score&sortProperties=cm%3Amodified&sortAscending=false&sortAscending=false&propertyFilter=-all-&`,
+			uri: `${process.env.ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/`,
+			qs: {
+				contentType,
+				skipCount,
+				sortProperties,
+				sortAscending,
+				propertyFilter,
+			},
+
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] }
+				],
+				facettes: ['cclom:general_keyword']
+			}),
+			timeout: REQUEST_TIMEOUT
+		}; */
+
+		const eduResponse = await request(options);
+		let parsed = JSON.parse(eduResponse);
+
+		// provided by client eg data.query.filterOptions
+		const filterOptions = {
+			mimetype: ['text/html', 'image/jpeg'],
+			provider: ['BauhausMaterial.de', 'München educationcenter', 'Khan Academy']
+		}
+
+		// checks if user has set filter options
+		if (!!Object.values(filterOptions).length) {
+			parsed = filterResult(parsed, filterOptions)
+		}
+
+		function filterResult(obj, options) {
+			let result;
+			// checks if user has set type filter
+			if (options.mimetype.length) {
+				result = filterMime(obj, options.mimetype)
+			}
+			// checks if user has set provider filter
+			if (options.provider.length) {
+				result = filterProvider(obj, options.provider)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['filterProvider' was used before it was defined.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4791331)